src/plugins/credential_management/public/components/edit_credential_wizard/components/edit_credential.tsx

```diff
@@ -38,21 +46,25 @@ import { CredentialManagmentContextValue } from '../../../types';
 import { context as contextType } from '../../../../../opensearch_dashboards_react/public';
 import { EditCredentialItem } from '../../types';
 import { LocalizedContent } from '../../common/text_content';
+import { EditPageHeader } from '../../common/components/header/edit_page_header';
 
 interface EditCredentialState {
   credentialName: string;
   credentialMaterialsType: string;
+  credentialDescription: string;
   username?: string;
   password?: string;
   dual: boolean;
   toasts: EuiGlobalToastListToast[];
   docLinks: DocLinksStart;
-  isVisible: boolean;
+  isDeleteModalVisible: boolean;
+  isUpdateModalVisible: boolean;
   isLoading: boolean;
 }
 
 export interface EditCredentialProps extends RouteComponentProps {
   credential: EditCredentialItem;
+  originalCredential: EditCredentialItem;
 }
 
 export class EditCredentialComponent extends React.Component<
```

Comment:
Can we add more comments for readability, also make the file more orgranized? For example, group functions into validation / events / render methods?